libs/execution/src/lib/execution-context.ts

```diff
@@ -129,9 +131,14 @@ export class ExecutionContext {
     assert(!isPipelineDefinition(currentNode));
     assert(!isExpressionConstraintDefinition(currentNode));
     assert(!isTransformDefinition(currentNode));
-    assert(isReference(currentNode.type));
 
-    const metaInf = getOrFailMetaInformation(currentNode.type);
+    let metaInf: MetaInformation;
+    if (isConstraintDefinition(currentNode)) {
+      metaInf = getOrFailMetaInformation(currentNode.type);
+    } else {
+      assert(isReference(currentNode.type));
+      metaInf = getOrFailMetaInformation(currentNode.type);
+    }
```

Comment:
Can u do sth like 
```
if (isBlockDefinition(currentNode)) {
  assert(isReference(currentNode.type);
}
const metaInf = getOrFailMetaInformatin(currentNode.type);
```

Would be a little easier to read.

Reply:
That unfortunately does not work, because it would require a declaration of the following type: `getOrFailMetaInformation(type: Reference<BuiltinBlocktypeDefinition> | ConstraintTypeLiteral)`

The reason being that Typescript is unable to detect multiple declarations and decide which one to use.